content/blog/zardshard/2023-05-23_how_to_find_memory_leaks.md

```diff
@@ -0,0 +1,59 @@
++++
+type = "blog"
+title = "How to find memory leaks"
+author = "Zardshard"
+date = "2023-05-23 10:44:38-04:00"
+tags = ["malloc_debug", "libroot_debug.so", "heap", "memory", "debugging"]
++++
+
+In 2010-2011, mmlr created a new memory allocator: the guarded heap memory allocator. This allocator help detect various bugs such as writing past the end of allocated memory, reading uninitialized memory, and freeing freed memory. These uses are detailed in ["Using malloc_debug to Find Memory Related Bugs"](https://www.haiku-os.org/blog/mmlr/2010-02-08_using_malloc_debug_find_memory_related_bugs/). Later, in 2015, mmlr had a new project: updating the memory allocator to be able to report memory leaks.
```

Comment:
This allocator help detect (helps)

Reply:
Good catch! I ran it through a grammar checker and found two more errors.